### PR TITLE
Add functions to `dict`

### PR DESCRIPTION
**Pull request by samuelchassot**

### CODE REVIEW

File: pyfuncol/dict.py [hunk 1]
```diff
@@ -107,6 +107,124 @@ def to_list(self: Dict[A, B]) -> List[Tuple[A, B]]:
...
     return [(k, v) for k, v in self.items()]
 
+def count(self: Dict[A, B], p: Callable[[Tuple[A, B]], bool]) -> int:
+    """
+    Counts the number of elements in the collection which satisfy a predicate.
+
+    Note: will not terminate for infinite-sized collections.
+
+    Args:
+        p: the predicate used to test elements.
```
Comment (Gondolav) [inline pyfuncol/dict.py:117]:
```suggestion
        p: The predicate used to test elements.
```

File: pyfuncol/dict.py [hunk 2]
```diff
@@ -107,6 +107,124 @@ def to_list(self: Dict[A, B]) -> List[Tuple[A, B]]:
...
+    
+    Returns:
+        the number of elements satisfying the predicate p.
```
Comment (Gondolav) [inline pyfuncol/dict.py:120]:
```suggestion
        The number of elements satisfying the predicate p.
```

File: pyfuncol/dict.py [hunk 3]
```diff
@@ -107,6 +107,124 @@ def to_list(self: Dict[A, B]) -> List[Tuple[A, B]]:
...
+def fold_left(self: Dict[A, B], z: B, op: Callable[[B, Tuple[A, B]], B]) -> B:
+    """
+    Applies a binary operator to a start value and all elements of this collection, going left to right.
+
+    Note: will not terminate for infinite-sized collections.
+
+    Note: might return different results for different runs, unless the underlying collection type is ordered or the operator is associative and commutative.
+
+    Args:
+        z: the start value.
```
Comment (Gondolav) [inline pyfuncol/dict.py:138]:
```suggestion
        z: The start value.
```

File: pyfuncol/dict.py [hunk 4]
```diff
@@ -107,6 +107,124 @@ def to_list(self: Dict[A, B]) -> List[Tuple[A, B]]:
...
+        op: the binary operator.
```
Comment (Gondolav) [inline pyfuncol/dict.py:139]:
```suggestion
        op: The binary operator.
```

File: pyfuncol/dict.py [hunk 5]
```diff
@@ -107,6 +107,124 @@ def to_list(self: Dict[A, B]) -> List[Tuple[A, B]]:
...
+    
+    Returns:
+        the result of inserting op between consecutive elements of this collection, going left to right with the start value z on the left:
```
Comment (Gondolav) [inline pyfuncol/dict.py:142]:
```suggestion
        The result of inserting op between consecutive elements of this collection, going left to right with the start value z on the left:
```

**Review by Gondolav — CHANGES_REQUESTED** [review]
Perfect, just a few fixes and a black refactoring pass and it's ready!

**Comment by samuelchassot** [comment]
whoops wasn't ready for review haha but will be soon

**Review by Gondolav — APPROVED** [review]

**Pull request merged**